src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/CounterfactualLogger.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading;
+using NuGet.Common;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    /// <summary>
+    /// Keeps all counterfactuals state in one place
+    /// </summary>
+    internal class CounterfactualLogger
+    {
+        private int _emittedFlag = 0;
+
+        internal CounterfactualLogger(string eventName)
+        {
+            EventName = eventName + "Counterfactual";
+        }
+
+        internal string EventName { get; }
+
+        /// <summary>
+        /// Emits counterfactual telemetry event once
+        /// </summary>
+        public void TryEmit()
+        {
+            if (Interlocked.CompareExchange(ref _emittedFlag, 1, 0) == 0)
+            {
+                try
```

Comment:
I think we should trust the Telemetry service will work as expected and not try to account for the telemetry service throwing. 

When thinking about scorecard analysis, it's hard to argue that some telemetry is more important than others. 
Even if you could guarantee this event is being sent here, what about other events? What if this experiment is causing events in a totally different place to fail 50% of the time, and we're using them in our scorecard analysis? 
Sure you could say, we can validate the event totals based on an exp, but that becomes a slippery slope.

Reply:
Think of it as an extra caution measure.

Reply:
> Think of it as an extra caution measure.

I'm worried about what kind of precedent this is setting. 

If tomorrow I create a PR, would someone ping me to wrap every telemetry emission in a try catch? 
It's overkill.

Reply:
Removed.